cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -981,6 +984,14 @@ public void createOrDeleteReplicas(Map<Short, Set<String>> topicsByReplicationFa
         int[] cursors = new int[racks.size()];
         int rackCursor = 0;
         for (PartitionInfo partitionInfo : cluster.partitionsForTopic(topic)) {
+
+          Partition partition = partition(new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
+          if (!hasSameReplicasFor(partition, partitionInfo)) {
+            LOG.warn("Detected partition info inconsistent with clusterModel: PartitionInfo: {}, Partition in ClusterModel: {}. "
+                     + " Skip creating or deleting replicas for this partition.", partitionInfo, partition);
+            continue;
+          }
+
```

Comment:
Since this if check is more expensive compared to the following if check (i.e. `if (partitionInfo.replicas().length == replicationFactor) {`), can we move that one before this check?